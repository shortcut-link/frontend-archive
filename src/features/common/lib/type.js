//@flow
export type Method = 'GET' | 'POST' | 'PUT' | 'DELETE';

export type Options = {
  headers?: { [key: string]: string },
  parse?: 'text' | 'json' | 'noparse',
  baseURI?: string,
  body?: string | FormData | mixed
};

export type ResponseOk<R> = {
  ok: true,
  result: R
};

export type ResponseError<E> = {
  ok: false,
  error: E
};

export type CustomResponse<R, E> = ResponseOk<R> | ResponseError<E>;
