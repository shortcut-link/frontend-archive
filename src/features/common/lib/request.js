//@flow
import { $token } from '../model/token';
import type { Method, Options, CustomResponse } from './type';

const baseURI = '/api';

export const request = (
  method: Method,
  url: string,
  options: Options = {}
): Promise<any> => {
  const token = $token.getState();

  const headers = new Headers({
    ...createContentType(options),
    ...createAuthorization(token),
    ...options.headers
  });

  const uri = `${options.baseURI || baseURI}${url}`;

  const { body, ...restOptions } = options;

  const config = new Request(uri, {
    method,
    headers,
    ...restOptions,
    body: createBody(options, headers)
  });

  return fetch(config).then(response => {
    if (options.parse === 'text') {
      return response.text;
    }

    if (options.parse === 'noparse') {
      return response.text;
    }

    const contentType = response.headers.get('Content-Type');

    if (contentType && contentType.includes('json')) {
      return response.json().then(responseToPromise, responseToPromise);
    }

    throw new TypeError('Unexpected content-type');
  });
};

const createContentType = (options: Options) => {
  const header = contentTypeFromOptions(options);

  return header ? { 'Content-Type': header } : {};
};

const contentTypeFromOptions = (options: Options) => {
  if (options && options.headers && options.headers['Content-Type']) {
    return options.headers['Content-Type'];
  }

  if (options && options.body && options.body instanceof FormData) {
    return 'multipart/form-data';
  }

  return typeof options.body === 'object' ? 'application/json' : '';
};

const createAuthorization = (token: ?string) =>
  token ? { Authorization: `bearer ${token}` } : {};

const createBody = (
  options: Options,
  headers: Headers
): FormData | string | void => {
  const contentType = headers.get('content-type');

  if (options.body && contentType && contentType.includes('json')) {
    return JSON.stringify(options.body);
  }

  if (options.body instanceof FormData) {
    return options.body;
  }

  return undefined;
};

function responseToPromise<R, E>(response: CustomResponse<R, E>) {
  return response && typeof response.ok === 'boolean'
    ? okToPromise<R, E>(response)
    : response;
}

function okToPromise<R, E>(response: CustomResponse<R, E>): Promise<R> {
  return response.ok
    ? Promise.resolve(response.result)
    : Promise.reject(response.error);
}
