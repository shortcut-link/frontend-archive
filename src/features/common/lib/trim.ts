import { SyntheticEvent } from 'react';

export const trimEvent = (event: SyntheticEvent<HTMLInputElement>) =>
  event.currentTarget.value.trim();
