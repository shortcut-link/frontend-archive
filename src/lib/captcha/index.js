import React from 'react';
import ReCAPTCHA from 'react-google-recaptcha';

const key = '6LdPb7MUAAAAAEEVmEYPQq9kE6TzPPgfcBwua57l';

/**
 * reCaptcha
 * @param {{ onChange: function }} onChange Callback that is called when passing a captcha
 */
export const Captcha = ({ onChange }) => (
  <ReCAPTCHA sitekey={key} onChange={onChange} />
);
