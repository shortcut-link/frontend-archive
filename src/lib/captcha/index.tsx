import React from 'react';
import ReCAPTCHA from 'react-google-recaptcha';

const key = '6LdPb7MUAAAAAEEVmEYPQq9kE6TzPPgfcBwua57l';

interface CaptchaProps {
  onChange: (payload: void) => void;
}

export const Captcha: React.FC<CaptchaProps> = ({ onChange }) => (
  <ReCAPTCHA sitekey={key} onChange={token => onChange()} />
);
