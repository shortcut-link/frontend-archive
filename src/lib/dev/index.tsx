import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { BrowserRouter } from 'react-router-dom';

import { lightTheme } from 'ui/themes';

/**
 * For testing components that require ThemeProvider or BrowserRouter
 */
export const ThemeRouterProvider = (component: React.ReactNode) =>
  renderer.create(
    <ThemeProvider theme={lightTheme}>
      <BrowserRouter>{component}</BrowserRouter>
    </ThemeProvider>
  );
