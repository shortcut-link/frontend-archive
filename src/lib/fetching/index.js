import { createStore, createEvent } from 'effector';

export function createFetching(effect, initialStatus = 'initial', params = {}) {
  const customReset = params.reset || createEvent();

  const result = createStore(params.result || null)
    .reset(effect)
    .reset(effect.fail)
    .reset(customReset)
    .on(effect.done, (_, { result: value }) => value);

  const error = createStore(params.error || null)
    .reset(effect)
    .reset(effect.done)
    .reset(customReset)
    .on(effect.fail, (_, { error: value }) => value);

  const status = createStore(initialStatus)
    .on(effect, () => 'loading')
    .on(effect.done, () => 'done')
    .on(effect.fail, () => 'fail')
    .on(customReset, () => 'initial');

  const isDone = status.map(state => state === 'done');
  const isFailed = status.map(state => state === 'fail');
  const isLoading = status.map(state => state === 'loading');

  return { result, error, status, isDone, isFailed, isLoading };
}
