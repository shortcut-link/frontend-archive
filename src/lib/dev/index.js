import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { ToggleThemeProvider } from 'lib/theme';
import { lightTheme, darkTheme } from 'ui/themes';

/* For testing components that
 * require ThemeProvider or BrowserRouter
 */

export const ThemeProviderRouter = ({ children }) => (
  <ToggleThemeProvider light={lightTheme} dark={darkTheme}>
    <BrowserRouter>{children}</BrowserRouter>
  </ToggleThemeProvider>
);
