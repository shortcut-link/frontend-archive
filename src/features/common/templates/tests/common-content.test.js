import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { CommonContentTemplate } from '../index';
import { ThemeProviderRouter } from 'lib/dev';

describe('must render Header', () => {
  const children = <div>Children</div>;

  it('with default header, footer', () => {
    const component = (
      <ThemeProviderRouter>
        <CommonContentTemplate>{children}</CommonContentTemplate>
      </ThemeProviderRouter>
    );

    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('with custom header, footer', () => {
    const header = <div>Header</div>;
    const footer = <div>Footer</div>;
    const component = (
      <ThemeProviderRouter>
        <CommonContentTemplate header={header} footer={footer}>
          {children}
        </CommonContentTemplate>
      </ThemeProviderRouter>
    );

    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
