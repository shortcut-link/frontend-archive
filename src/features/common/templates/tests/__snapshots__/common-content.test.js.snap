// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must render Header with custom header, footer 1`] = `
.c0 {
  display: grid;
  min-height: 100vh;
  max-height: 100vh;
  grid-template-rows: auto 1fr auto;
  grid-template-areas: 'header' '.' 'footer';
}

.c1 {
  grid-area: header;
}

.c4 {
  grid-area: footer;
}

.c3 {
  -webkit-box-pack: initial;
  -webkit-justify-content: initial;
  -ms-flex-pack: initial;
  justify-content: initial;
  -webkit-align-items: initial;
  -webkit-box-align: initial;
  -ms-flex-align: initial;
  align-items: initial;
  -webkit-flex-wrap: initial;
  -ms-flex-wrap: initial;
  flex-wrap: initial;
  width: initial;
  min-width: initial;
  max-width: initial;
  padding: initial;
  display: inherit;
  max-width: 50rem;
  width: 100%;
  padding: 1rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #ECEFF1;
  color: Black;
}

<div
  className="c0"
>
  <header
    className="c1"
  >
    <header>
      Header
    </header>
  </header>
  <div
    className="c2"
  >
    <div
      className="c3"
    >
      <div>
        Children
      </div>
    </div>
  </div>
  <footer
    className="c4"
  >
    <footer>
      Footer
    </footer>
  </footer>
</div>
`;

exports[`must render Header with default header, footer 1`] = `
.c0 {
  display: grid;
  min-height: 100vh;
  max-height: 100vh;
  grid-template-rows: auto 1fr auto;
  grid-template-areas: 'header' '.' 'footer';
}

.c1 {
  grid-area: header;
}

.c3 {
  -webkit-box-pack: initial;
  -webkit-justify-content: initial;
  -ms-flex-pack: initial;
  justify-content: initial;
  -webkit-align-items: initial;
  -webkit-box-align: initial;
  -ms-flex-align: initial;
  align-items: initial;
  -webkit-flex-wrap: initial;
  -ms-flex-wrap: initial;
  flex-wrap: initial;
  width: initial;
  min-width: initial;
  max-width: initial;
  padding: initial;
  display: inherit;
  max-width: 50rem;
  width: 100%;
  padding: 1rem;
}

.c6 {
  position: relative;
  display: block;
  width: calc((1.4rem * 2) + 8px);
  height: calc(1.4rem + 8px);
}

.c10 {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: rgba(0,0,0,0.12);
  border-radius: 1.4rem;
  cursor: pointer;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  background-color: #64B5F6;
}

.c10:before {
  content: "";
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  top: 4px;
  left: 4px;
  width: 1.4rem;
  height: 1.4rem;
  font-size: calc(1.4rem / 2);
  line-height: 0px;
  background-color: white;
  border-radius: 50%;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.c10:before {
  content: '☀️';
}

.c7 {
  opacity: 0;
  width: 0;
  height: 0;
}

.c7:checked {
  -webkit-transform: translateX(1.4rem);
  -ms-transform: translateX(1.4rem);
  transform: translateX(1.4rem);
}

.c7:checked + .c8 {
  background-color: #64B5F6;
}

.c7:checked + .c8::before {
  -webkit-transform: translateX(1.4rem);
  -ms-transform: translateX(1.4rem);
  transform: translateX(1.4rem);
}

.c7:checked + .c9::before {
  content: '🌙';
}

.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: 0.15s;
  transition: 0.15s;
  color: inherit;
  font-size: inherit;
  outline: none;
  -webkit-box-pack: initial;
  -webkit-justify-content: initial;
  -ms-flex-pack: initial;
  justify-content: initial;
  -webkit-align-items: initial;
  -webkit-box-align: initial;
  -ms-flex-align: initial;
  align-items: initial;
  -webkit-flex-wrap: initial;
  -ms-flex-wrap: initial;
  flex-wrap: initial;
  width: initial;
  min-width: initial;
  max-width: initial;
  padding: initial;
}

.c4:hover,
.c4:focus {
  color: #2196F3;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 > :not(:first-child) {
  margin-left: 20px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 5rem;
  background-color: #ECEFF1;
  color: Black;
}

.c2 > div.container {
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #ECEFF1;
  color: Black;
}

<div
  className="c0"
>
  <header
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="container c3"
      >
        <h1>
          <a
            className="c4"
            href="/"
            onClick={[Function]}
            style={
              Object {
                "textTransform": "uppercase",
              }
            }
          >
            Shortcut-link
          </a>
        </h1>
        <div
          className="c5"
        >
          <a
            className="c4"
            href="/join"
            onClick={[Function]}
          >
            Login to account
          </a>
          <label
            className="c6"
            onKeyPress={[Function]}
            radius="1.4rem"
          >
            <input
              checked={false}
              className="c7"
              onChange={[Function]}
              radius="1.4rem"
              type="checkbox"
              value={false}
            />
            <span
              className="c8 slider c9 c10"
              radius="1.4rem"
            />
          </label>
        </div>
      </div>
    </div>
  </header>
  <div
    className="c11"
  >
    <div
      className="c3"
    >
      <div>
        Children
      </div>
    </div>
  </div>
</div>
`;
